### PR TITLE
contrib: require an end-user sponsor

### PR DESCRIPTION
**Pull request by mattklein123**

### CODE REVIEW

**Comment by mattklein123** [comment]
cc @envoyproxy/envoy-maintainers @moderation for comment

File: EXTENSION_POLICY.md [hunk 1]
```diff
@@ -147,6 +147,14 @@ included by default in the main image builds. Consumers need to pull directly fr
 images described in the installation guide. Please read the linked document in detail to determine
 whether contrib extensions are the right choice for a newly proposed extension.
 
+**NOTE:** Contrib extensions **require** an end-user sponsor. The sponsor is someone who will run
```
Comment (mattklein123) [inline EXTENSION_POLICY.md:150]: Note: need to further define end-user and probably cross link to the security process which also talks about end-users. We should use the same definition in both places. cc @moderation

Comment (mattklein123) [reply EXTENSION_POLICY.md:150]: Referenced https://github.com/envoyproxy/envoy/blob/main/SECURITY.md#membership-criteria but not really well defined. So let's carefully define and then cross link.

Very roughly, and end-user is someone that uses Envoy to directly serve traffic for a 1st party offering. A vendor can also be an end-user (Google would be the canonical example of this within the community.)

**Review by jmarantz — APPROVED** [review]
This makes sense to me. Thanks for writing this up.

**Review by ggreenway — APPROVED** [review]

**Review by phlax — APPROVED** [review]

**Review by kyessenov — DISMISSED** [review]

**Review by adisuissa — APPROVED** [review]

**Review by wrowe — DISMISSED** [review]

**Review by alyssawilk — APPROVED** [review]

**Comment by mattklein123** [comment]
I'm going to take a stab at better defining end user today and will ping back for another review.

**Comment by mattklein123** [comment]
@envoyproxy/envoy-maintainers I updated this PR and attempted to define end user which also is relevant to the security list membership policy. PTAL.

**Review by alyssawilk — APPROVED** [review]

**Review by KBaichoo — APPROVED** [review]

**Pull request merged**